backend/routers/voice.py

```
@@ -0,0 +1,321 @@
+    
+    - **audio_file**: Audio file to transcribe
+    - **preferred_language**: Preferred language code ('auto', 'hi', 'mr', 'en', etc.)
+    
+    Returns transcribed text, translated text (English), and confidence score
+    """
+    try:
+        # Read audio file
+        audio_content = await audio_file.read()
+        
+        if not audio_content:
+            raise HTTPException(status_code=400, detail="Empty audio file provided")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)